### PR TITLE
chore: [CO-988] use project mvn revision property to generate build info class

### PR DESCRIPTION
**Pull request by keshavbhatt**
**What has changed:**

Project revision property set in project `pom.xml` is shared with ant task that generate carbonio-build-info(`BuildInfoGenerated`) classes.

This omits the need to update the `ant-build-common.xml` during release process.

**Bonus:**

- Added SOAP API changelog generation ant tasks. 
    - `generate-soap-api-changelog` to generate changelog comparing the baseline api description json.
    - `update-baseline-apidesc` to update the existing baseline api description json
    - Updated the changelog generation template and style.

- Simplification and cleanup of related code and instrumentation.

### CODE REVIEW

File: ant-build-common.xml [hunk 1]
```diff
@@ -62,8 +62,9 @@
     <echo level="info" message="Using version ${zimbra.buildinfo.version}"/>
   </target>
 
+  <property file="${maven.build.dir}/mvn-ant-shared-build.properties"/>
   <target name="carbonio-version">
-    <property name="carbonio.buildinfo.version" value="24.4.0_ZEXTRAS_202404"/>
+    <property name="carbonio.buildinfo.version" value="${mvn_revision}"/>
```
Comment (frisonisland) [inline ant-build-common.xml:67]: Just a question: what is the value of mvn_revision?
I see that mvn-ant-shared-build.properties is generated with mvn_revision value equal to revision, but revision is 24.4.0 and we miss the _ZEXTRAS_202404 part. Maybe we do not need it anymore?

Comment (keshavbhatt) [reply ant-build-common.xml]: It is needed, it is used as build timestamp in generated build info. 
I provided alternative solution that uses current datetime YYYYMM format.

File: store/src/main/java/com/zimbra/cs/util/BuildInfo.java [hunk 2]
```diff
@@ -32,15 +31,18 @@ public final class BuildInfo {
 
     public static final String FULL_VERSION;
 
+    public static final String UNKNOWN = "unknown";
```
Comment (frisonisland) [inline store/src/main/java/com/zimbra/cs/util/BuildInfo.java:34]: Just a suggestion that we can make this private to avoid exposing things outside

Comment (keshavbhatt) [reply store/src/main/java/com/zimbra/cs/util/BuildInfo.java:34]: sure

**Review by matteobaglini — APPROVED** [review]

**Pull request merged**